cpp/include/cudf/ast/ast.cuh

```diff
@@ -0,0 +1,436 @@
+/*
```

Comment:
All of the kernels and such should be moved to a detail header somewhere. The public header should only expose the `compute_column` and any other symbols it depends on.

Reply:
Fixed. Only `compute_column` is exposed in `cudf/ast/transform.hpp` (new, more informative name).